Source/JavaScriptCore/jit/JITArithmetic.cpp

```diff
@@ -273,18 +332,43 @@ void JIT::emit_compareUnsignedImpl(VirtualRegister dst, VirtualRegister op1, Vir
     emitPutVirtualRegister(dst, jsRegT10);
 }
 
+template<typename Op, typename SlowOperation>
+void JIT::emit_compareSlow(const JSInstruction* instruction, DoubleCondition condition, SlowOperation operation, Vector<SlowCaseEntry>::iterator& iter)
+{
+    auto bytecode = instruction->as<Op>();
+    VirtualRegister dst = bytecode.m_dst;
+    VirtualRegister op1 = bytecode.m_lhs;
+    VirtualRegister op2 = bytecode.m_rhs;
+    auto handleReturnValueGPR = [&]() {
+        boxBoolean(returnValueGPR, jsRegT10);
+        emitPutVirtualRegister(dst, jsRegT10);
```

Comment:
Let's ensure that the result of the operation is 1 or 0 (not something 42 is returned for example).